### ISSUE TITLE
HPND License is Deprecated

### ISSUE THREAD

**Issue by scotgopal** [python-pillow/Pillow#5975]
Hello there. I'm reporting this as I found that the HPND License, which this library is licensed with, had been [deprecated by its author](https://opensource.org/licenses/HPND). I'm not really sure when though.

Hence as a reaction, I would recommend relicensing this library under MIT License or Apache License 2.0. 😄 

Ps: You may consider checking out https://choosealicense.com/ to help make decision on the new license.

![image](https://user-images.githubusercontent.com/76937732/150066048-e197d194-fe9f-40f3-a1db-8d13feb3292e.png)

**Comment by radarhere** [comment]
If the wayback machine is to be believed, then that notice has actually been there since 2011 - https://web.archive.org/web/20111021125604/https://opensource.org/licenses/HPND

**Comment by hugovk** [comment]
I'm not a lawyer: it has been deprecated and it's unlikely we'll use it for any new projects.

But I believe the old licence is still valid, and to change it would require getting permission from all the past contributors listed at https://github.com/python-pillow/Pillow/graphs/contributors, plus all the contributors from the original PIL who are not listed there. Alternatively we'd require a dual licence approach, and/or rewriting and/or removing large chunks of the code.

Is the HPND licence a problem for you?

**Comment by scotgopal** [comment]
> I'm not a lawyer: it has been deprecated and it's unlikely we'll use it for any new projects.
> 
> But I believe the old licence is still valid, and to change it would require getting permission from all the past contributors listed at https://github.com/python-pillow/Pillow/graphs/contributors, plus all the contributors from the original PIL who are not listed there. Alternatively we'd require a dual licence approach, and/or rewriting and/or removing large chunks of the code.
> 
> Is the HPND licence a problem for you?

The licensing is not an issue for me. I was just looking out to make sure the contributors to this repo are protected. I have a few minor doubts with your suggestions. Wouldn't the dual license approach require getting permission from the past contributors as well? Also, by large chunks of code, do you mean the licensing part of a file alone or some other parts too?

Honestly, I'm not well-verse with licensing nor the legal repercussions of changing an open source license, but I wonder if it is okay to just replace a permissible license like HPND with other permissible types. I don't think reminding every past contributor is a practical thing to do as it is not a small number. Even if we want to, I'm not sure if there's a designated channel to reach out to them all at once. (too painful to even imagine emailing everyone of them and keep a record of their response). And I also believe that more recent open source licenses may provide much better protection and structure than older ones, which is safe to assume that every past contributor will agree to. Hence, I'd strongly suggest in swapping the old license with a modern one. We can always get more expert opinions from other open source contributors who knows these things better.

> Historical Permission Notice and Disclaimer (#HPND)
This is a lax, permissive, and weak free software license that is compatible with the GPL. It is similar to the license of Python 1.6a2 and earlier versions.

_[Source](https://www.gnu.org/licenses/license-list.en.html#HPND)_

Here's an additional link that maybe useful for future discussions and actions:-
1. [Full license text](https://directory.fsf.org/wiki/License:HPND)

**Comment by hugovk** [comment]
> The licensing is not an issue for me. 

Good to hear!

> I was just looking out to make sure the contributors to this repo are protected. I have a few minor doubts with your suggestions. Wouldn't the dual license approach require getting permission from the past contributors as well? Also, by large chunks of code, do you mean the licensing part of a file alone or some other parts too?

Dual licensing would require something like using the old licence for code authored by people who don't give permission to re-license their code, and only the new one for code where they do give permission. But even finding out would be a tangled mess, for no real benefit.

> Honestly, I'm not well-versed with licensing nor the legal repercussions of changing an open source license, but I wonder if it is okay to just replace a permissible license like HPND with other permissible types. 

No, we can't really just swap it out. https://github.com/python-pillow/Pillow/issues/1507#issuecomment-476702268:

> > If you were starting a new project, or a rewrite, I'd recommend a newer license, like BSD-2-clause+Patent.
> >
> > But since you've got all this old code contributed by people who are no longer with the project, you're pretty much stuck with the existing license - you can't simply make it go away. So not much upside in switching for new contributions; then people just have to comply with two licenses.

.

> I don't think reminding every past contributor is a practical thing to do as it is not a small number. 

Agreed!

> And I also believe that more recent open source licenses may provide much better protection and structure than older ones, which is safe to assume that every past contributor will agree to. 

Really? Whilst we'd choose a modern one for a new project, I don't think HPND is so different from some modern ones either. And we can't assume what hundreds of people might agree to without asking them.

> Hence, I'd strongly suggest in swapping the old license with a modern one. We can always get more expert opinions from other open source contributors who knows these things better.

It would be such a big undertaking to change it, plus the added work for downstream users and packagers, we would need to hear some strong, concrete reasons up front.

(PS HPND is also mentioned here: https://www.fsf.org/blogs/licensing/historical-permission-notice-and-disclaimer-added-to-license-list )

But thanks again for raising!

**Comment by wiredfool** [comment]
At this point, given that we've got some indication that this license was supposed to be the old python license with trivial modifications from effbot (which is not responding any more, RIP), and we've got a lawyer's internet opinion that the license is ok enough (https://github.com/python-pillow/Pillow/issues/1507#issuecomment-476702268), I'd say that for practical purposes, we're at the point of a MIT/BSD style license.  The only licenses that we could potentially move to would be the python license, or MIT/BSD, which wouldn't gain us anything other than a bit of clarity. 

I think any relicensing would at the very least have to have the agreement of the vast majority of the contributors, one of which would be the estate of Fredrik. This is not something that I'd want to bother them with.

I think this would be a lot of work for minimal practical gain, with the potential downside of legal cans of worms in multiple jurisdictions.
